internal/manager/manager.go

```diff
@@ -187,3 +193,88 @@ func (o *ObservedResources) Len() int {
 	defer o.mu.RUnlock()
 	return len(o.observed)
 }
+
+type kubeResource interface {
+	runtime.Object
+	metav1.Object
+}
+
+type resourceManager[R kubeResource] interface {
+	Create(ctx context.Context, obj R) (R, error)
+}
+
+// RevertUserInitiatedDeletion detects if a resource deletion was unauthorized and recreates the resource.
+// Returns true if the resource was recreated, false otherwise.
+func RevertUserInitiatedDeletion[R kubeResource](ctx context.Context,
```

Comment:
We can return errors from this function, rather than just logging them. 
- It's true that some (or all) of the calling functions might themselves just be logging them, but it at least opens up the possibility for handling the error by callers.

WDYT?